Source/WebKit/Shared/JavaScriptEvaluationResult.cpp

```diff
@@ -264,41 +272,91 @@ Protected<JSValueRef> JavaScriptEvaluationResult::toJS(JSGlobalContextRef contex
+    auto value = toVariant(variant);
+    auto identifier = JSObjectID::generate();
+    m_map.add(identifier, WTFMove(value));
+    return identifier;
 }
 
-Protected<JSValueRef> JavaScriptEvaluationResult::toJS(JSGlobalContextRef context)
+auto JavaScriptEvaluationResult::toVariant(GVariant* variant) -> Value
 {
-    Ref serializedScriptValue = API::SerializedScriptValue::createFromWireBytes(wireBytes());
-    return Protected<JSValueRef>(context, serializedScriptValue->internalRepresentation().deserialize(context, nullptr));
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE("a{sv}"))) {
+        HashMap<JSObjectID, JSObjectID> map;
+        GVariantIter iter;
+        g_variant_iter_init(&iter, variant);
+        const char* key;
+        GVariant* value;
+        while (g_variant_iter_loop(&iter, "{&sv}", &key, &value)) {
+            if (!key || !value)
+                continue;
+            auto keyID = JSObjectID::generate();
+            m_map.add(keyID, String::fromUTF8(key));
+            auto valueID = JSObjectID::generate();
+            m_map.add(valueID, toVariant(value));
+            map.add(keyID, valueID);
+        }
+        return { WTFMove(map) };
+    }
+
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_UINT32))
+        return (double)g_variant_get_uint32(variant);
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_INT32))
+        return (double)g_variant_get_int32(variant);
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_UINT64))
+        return (double)g_variant_get_uint64(variant);
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_INT64))
+        return (double)g_variant_get_int64(variant);
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_INT16))
+        return (double)g_variant_get_int16(variant);
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_UINT16))
+        return (double)g_variant_get_uint16(variant);
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_DOUBLE))
+        return (double)g_variant_get_double(variant);
+
+    if (g_variant_is_of_type(variant, G_VARIANT_TYPE_STRING))
+        return String::fromUTF8(g_variant_get_string(variant, nullptr));
+
+    return EmptyType::Null;
 }
 
-JavaScriptEvaluationResult::JavaScriptEvaluationResult(JSGlobalContextRef context, JSValueRef value)
-    : m_valueFromJS(WebCore::SerializedScriptValue::create(context, value, nullptr))
-    , m_wireBytes(m_valueFromJS ? m_valueFromJS->wireBytes().span() : std::span<const uint8_t>())
+JavaScriptEvaluationResult::JavaScriptEvaluationResult(GVariant* variant)
+    : m_root(addObjectToMap(variant))
 {
 }
 
-std::optional<JavaScriptEvaluationResult> JavaScriptEvaluationResult::extract(JSGlobalContextRef context, JSValueRef value)
+std::optional<JavaScriptEvaluationResult> JavaScriptEvaluationResult::extract(GVariant* variant)
 {
-    return JavaScriptEvaluationResult { context, value };
+    if (!variant)
+        return std::nullopt;
+    return JavaScriptEvaluationResult(variant);
```

Comment:
The other problem is here, we need to check first if the variant can be serialized like cocoa does. I'll attach a patch with both fixes.